doc/htmldoc/developer_space/workflows/nest_with_ides.rst

```diff
@@ -410,7 +410,48 @@ Code Completion in Xcode
 We have to create a new target and configure it appropriately:
 
 1. In the menu select: File -> New -> Target....
-1. Make the target a OS X -> Command Line Tool (Next), of type C++ in your project (preselected). We call it ``completion``
-1. Remove all files and folders that are created with the new target.
-1. In the tab "Build Phase" of the new target, under "Compile Sources" add all ``*.h``, ``*.hpp``, ``*.c``, ``*.cc``, ``*.cpp`` files from the list (you can use CMD+a).
-1. Now Xcode generates its index and after that code completion should work.
+2. Make the target a OS X -> Command Line Tool (Next), of type C++ in your project (preselected). We call it ``completion``
+3. Remove all files and folders that are created with the new target.
+4. In the tab "Build Phase" of the new target, under "Compile Sources" add all ``*.h``, ``*.hpp``, ``*.c``, ``*.cc``, ``*.cpp`` files from the list (you can use CMD+a).
+5. Now Xcode generates its index and after that code completion should work.
+
+
+CLion
+-----
+
+This recipe describes how to set up CLion for editing, building, and running NEST. The description here was tested on Ubuntu 23.  
+All the steps below are fully optional if you don't intend to debug your code using CLion and prefer to build NEST manually from the command line.  
+If you just want to debug your code, you can skip steps 1-2 and only follow steps 3-7. However, it is advised to also follow steps 1-2 to inform CLion of your desired CMake configuration.  
+Setting up the CMake configuration in CLion has the following advantages:  
+
+* CLion's default CMake configuration creates the build directory inside the source code root directory, which easily leads to unknowingly committing build files to git.
+* CLion regenerates build files when checking out a new branch, potentially overriding the build you created by manually running CMake from the command line.
+* After setting up both the CMake configuaration and a run configuration, you won't need to use the command line to run ``cmake``, ``make``, ``make install``, and ``python *.py`` at all anymore. All will be done automatically by CLion and you only need to run your application without having to manually recompile.
+
```

Comment:
Please add this important bit of information:
```suggestion

.. note::

   CLion is a commercial product. It is *your responsibility* to ensure that you have a valid
   license permitting you to use CLion (or any software product) for your work on the    
   NEST Simulator. 
```

Reply:
Agreed, this is indeed an important note for all commercial products!